src/Explorer/Notebook/NotebookContainerClient.ts

```diff
@@ -66,48 +81,64 @@ export class NotebookContainerClient {
 
     const { notebookServerEndpoint, authToken } = this.getNotebookServerConfig();
     try {
-      if (this.checkStatus()) {
-        const response = await fetch(`${notebookServerEndpoint}api/metrics/memory`, {
-          method: "GET",
-          headers: {
-            Authorization: authToken,
-            "content-type": "application/json",
+      const runMemoryAsync = async () => {
```

Comment:
can just define it within promise retry if runMemoryAsync is not used anywhere else

Reply:
The method block is getting bigger and would be complex to read in a single go.